netbox/ipam/api/field_serializers.py

```diff
@@ -0,0 +1,38 @@
+from django.utils.translation import gettext_lazy as _
+from rest_framework import serializers
+
+from ipam import models
+from ipam.validators import validate_ipaddress_with_mask
+from netaddr import AddrFormatError, IPNetwork
+
+__all__ = [
+    'IPAddressField',
+]
+
+
+#
+# IP address field
+#
+
+class IPAddressField(serializers.CharField):
+    """IPAddressField with mask"""
+
+    default_error_messages = {
+        'invalid': _('Enter a valid IPv4 or IPv6 address with optional mask.'),
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        validator = validate_ipaddress_with_mask
+        self.validators.append(validator)
```

Comment:
I don't think we need to add the validator since we're already validating the address format under `to_internal_value()`. My impression is that the validators are more intended for the application of conditional rules (e.g. a string's max length) as opposed to the data format. The logic is redundant anyway: The validator doesn't provide any further validation beyond what you have under `to_internal_value()` already, so I think it's safe to ditch, unless the validator plays a role in generating the schema somehow.

Reply:
done